smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/AddHttpApiKeyAuthPlugin.java

```diff
@@ -142,6 +143,24 @@ public void writeAdditionalFiles(
                 });
     }
 
+    @Override
+    public void writeAdditionalExports(
+            TypeScriptSettings settings,
+            Model model,
+            SymbolProvider symbolProvider,
+            TypeScriptWriter writer
+    ) {
+        boolean isClientSdk = settings.generateClient();
+        ServiceShape service = settings.getService(model);
+        if (isClientSdk && hasEffectiveHttpApiKeyAuthTrait(model, service)) {
+            writer.write("export * from \"./middleware/$1L\";", getMiddlewarePath());
+        }
+    }
+
+    private String getMiddlewarePath() {
+        return "HttpApiKeyAuth";
+    }
```

Comment:
I think this is fair as a static constant, don't think it needs a private method